### DISCUSSION TITLE
Client template

### DISCUSSION THREAD

**Discussion by wolfclan68** [ltsp/ltsp#202] [Q&A]
Does anyone know a way to create a clone of a user account? I've Googled for it, but what I've found doesn't do it quite right. I plan to deploy workstations, the same hardware just replacing Windows 7, to various classrooms and would like them to be identical. In Active Directory with Windows, they could all be signed into the same account, but that doesn't work with LTSP. Thank you for any assistance.

NVM I Googled a little more and found this script that works quite nicely https://unix.stackexchange.com/questions/204970/clone-linux-user-copy-user-based-on-another-one

**Comment by enaut** [comment]
This sounds to me that you don't want a clone of a user but actually you want a guest user - no persistent files everything is reset after every login - else you will end up with a lot of lingering files and browser password store will be a security nightmare. So look at guest users discussed here:

 * https://github.com/ltsp/community/wiki/Guest-session
 * https://github.com/ltsp/community/issues/14

If you are done with the issue close it.

**Comment by wolfclan68** [comment]
@enaut I think that is a great idea, thank you. The first post you list is for the older version, I've attempted to follow it but it does not work for 19. Post ltsp/ltsp#178 is incomplete and goes off-topic multiple times and appears to complete in IRC. Is there somewhere I can get information on how to implement this guest configuration correctly? Perhaps @alkisg could please offer some direction. Possibly a guest configuration page in the Wiki? Thank you for your time.

**Comment by enaut** [comment]
The wikipage is for ltsp19 and was written by @Dur0k - describe your problem and maybe he can help or someone else - as alkisg moved you to the community feel free to discuss things ;)

I think so far both posts are the only information.

**Comment by alkisg** [comment]
wolfclan68, I think a good start is to answer *here* the questions asked in https://github.com/ltsp/community/issues/14#issuecomment-529093934.

"Template account" or "guest account" has different meaning for different persons, so you need to provide more details.

**Comment by wolfclan68** [comment]
I realize I started this discussion as a template account, but I am leaning toward the guest setup per enaut's suggestion. I like the idea in the wiki page, but when trying to implement I've completed all steps except when trying to run the createguests script I get the following:

remote@LTSP:~/Scripts$ sudo ./createguests.sh
mkdir: cannot create directory ‘/home/nfs’: File exists
adduser: To avoid problems, the username should consist only of
letters, digits, underscores, periods, at signs and dashes, and not start with
a dash (as defined by IEEE Std 1003.1-2001). For compatibility with Samba
machine accounts $ is also supported at the end of the username
passwd: user 'guest{192.168.17.30..250}' does not exist

Here's the script, just edited the ip scheme:
base="guest"
pass="guest"

mkdir /home/nfs

for ip in {192.168.17.30..250}; do

    user="${base}${ip}"

    adduser $user --disabled-password --gecos ""

    echo -e "$pass\n$pass\n" | sudo passwd $user

    mv "/home/${user}" /home/nfs/

    ln -s "nfs/${user}" "/home/${user}"

done

**Comment by wolfclan68** [comment]
Do you need autologin, i.e. to automatically connect to that user without showing lightdm at all?
**Autologin yes**
Do you need passwordless login, i.e. to be able to select "guest", press [enter], and login without using a password?
**Autologin and automatic relogin would be ideal**
Are files saved on logout? Are they cleared on login? Are they cleared on reboot?
**File saving would not be necessary**
Do you need an empty /home/guest, or a template /home/guest which you have customized?
**I would like to customize what they see, remove items from the Activity pane and set Chrome to autolaunch**

**Comment by alkisg** [comment]
Then you need only **one** guest account, but with one directory per client on the server:

https://github.com/ltsp/community/issues/14#issuecomment-535437189

**Comment by wolfclan68** [comment]
Thanks. How do I customize what they see. I've signed in to the guest account and modified it how I want, but it doesn't appear that way when they autologin.

**Comment by alkisg** [comment]
@Dur0k has created a very nice wiki page for [guest sessions](https://github.com/ltsp/community/wiki/Guest-session), but it would be much easier if it used a single account instead of multiple ones.

I can write the basic idea here, but someone needs to expand it and update the wiki with it. If @Dur0k finds this a good idea and wants to update his page, awesome, otherwise I'd suggest that you start a new "Kiosk session" page and write as much as you were able to do so far.

On the server, you should have a single "guest" account, with "/home/guest", and you should arrange it the way you want it; make the browser autostart or remove items from the menu etc; LTSP won't help you in customizing the user account, that's a task for desktop environment documentation.

Then on the server you run `sudo mkdir /home/nfs` and you create /etc/exports.d/local.exports with:

```shell
/home/nfs	*(rw,async,crossmnt,no_subtree_check,no_root_squash,insecure)
/home/guest	*(ro,async,crossmnt,no_subtree_check,no_root_squash,insecure)
```

And you run `sudo exportfs -ra` to make the changes available to the clients. Then you put these under [clients] in ltsp.conf:

```shell
POST_INIT_CREATE_DIRS="
mount server:/home/nfs /mnt
mkdir -p /mnt/$MAC_ADDRESS/home/guest
chown guest:guest /mnt/$MAC_ADDRESS/home/guest
umount /mnt
mkdir -p /etc/guest-template"
FSTAB_HOME="server:/home/nfs/$MAC_ADDRESS/home /home nfs defaults,nolock 0 0"
FSTAB_TEMPLATE="server:/home/guest /etc/guest-template nfs defaults,nolock 0 0"
```

This makes the server /home/guest available in read-only mode in the client /etc/guest-template,
while the /home/guest on the client actually maps to a different server directory for each client, /home/nfs/$MAC_ADDRESS/home

And on each login, you can run the rsync command that @Dur0k mentions in his wiki page.

**Comment by wolfclan68** [comment]
Hey there @alkisg, so I've followed your guide above and client boots fine with autologin. But I can't seem to get it to boot to my modified version of "guest". It does sign into the guest account but it looks like a brand new user account with the "What's new in Ubuntu" splash screen. How do I get the clients to boot the modified version of the guest account?
Thank you as always!

**Comment by alkisg** [comment]
It wasn't supposed to be a guide. It was supposed to be hints for someone else to write and maintain a guide! :)

It's best if guides are maintained by people that **use** them, with the help of LTSP developers, **not** by LTSP developers if they don't have that use case.

I think at this time you should start a kiosk wiki page, write all the things that you did there, so that I can then comment if I see something wrong.

**Comment by enoch85** [comment]
This is now done here: https://github.com/ltsp/community/wiki/Single-guest-session-over-NFS

**Reply by dgroos** [reply]
Current address for wiki page: [https://github.com/ltsp/ltsp/wiki/Single-guest-session-over-NFS](https://github.com/ltsp/ltsp/wiki/Single-guest-session-over-NFS)

**Comment by wolfclan68** [comment]
Thank you for your reply @alkisg and @enoch85 for your write-up.
